nunchaku/models/transformers/transformer_flux.py

```diff
@@ -180,6 +180,13 @@ def forward_layer_at(
 
         return encoder_hidden_states, hidden_states
 
+    def norm1(
+        self,
+        hidden_states: torch.Tensor,
+        emb: torch.Tensor
+    ) -> tuple[torch.Tensor, torch.Tensor, torch.Tensor, torch.Tensor, torch.Tensor]:
+        return self.m.norm1_forward(0, hidden_states, emb)
```

Comment:
Can we expose not only first layer but arbitrary layer too?
e.g., 
norm1 -> norm_one_at

Reply:
fyi: https://github.com/mit-han-lab/nunchaku/issues/233#issuecomment-2788320686

Reply:
We can of course expose every layer, and it would make more sense. But to keep compatibility with the non quantized model, we should keep the name `norm1`.

Regarding the index, this is something that confused me a lot. In the usual Flux model, one get a reference to desired block, and then calls a certain layer on it, something like `model.transformer_blocks[3].norm1`. It is my understanding that in the case of quantized models, one only gets a reference to the collection of all transformer blocks, and then the index of the layer to be called goes into the call, something like `model.transformer_blocks.norm1_at(3)`. Why is the API designed this way? It makes it more difficult to keep compatibility than if one had a reference to the precise block instead

Reply:
https://github.com/mit-han-lab/nunchaku/blob/56728d205f74a41d21f620755beab45772db5d90/nunchaku/models/transformers/transformer_flux.py#L318-L320

We wrap all transformer blocks internally.
(not sure) I guess it is exposed in this way to match compatibility with diffusers

Reply:
But you see, `self.transformer_blocks` is a `ModuleList` consisting of a single object. Then whenever you want to do inference on this object, you have to specify an index. This is where the Nunchaku implementation differs from TeaCache, and I am not sure how to reconcile them. The diffusers implementation would have a list of 19 modules in there